app/pages/Dashboard/Households/HouseholdActionButton/HouseholdActionButton.tsx

```diff
@@ -1,25 +1,63 @@
+import { useDeleteHouseholdMutation } from '@camp/data-layer';
+import { debug } from '@camp/debug';
+import { showNotification } from '@camp/design';
+import type { HouseholdKeys, HouseholdListItem } from '@camp/domain';
 import { MenuIcon } from '@camp/icons';
 import { messages } from '@camp/messages';
 import type { AppRoute, PathParams } from '@camp/router';
 import { Link } from '@camp/router';
 import { createTestAttr } from '@camp/test';
+import { isNull } from '@fullstacksjs/toolbox';
 import { ActionIcon, Menu } from '@mantine/core';
 
+import { openDeleteHouseholdModal } from '../DeleteHouseholdModal';
+import { householdActionIds as ids } from './HouseholdActionButton.ids';
+
 interface Props {
   to: AppRoute;
+  household?: HouseholdKeys & HouseholdListItem;
   params?: PathParams<AppRoute>;
   menuId?: string;
   menuButtonId?: string;
-  onDelete?: React.MouseEventHandler<HTMLButtonElement>;
 }
 
 export const HouseholdActionButton = ({
   menuId,
   menuButtonId,
   to,
   params,
-  onDelete,
+  household,
 }: Props) => {
+  const [deleteHousehold] = useDeleteHouseholdMutation();
+  const { id, name } = household!;
```

Comment:
we can make it required if it can not be null

Reply:
it can be null i guess.

Reply:
it can be null i guess.